packages/projection-typeorm/src/entity/BlockData.entity.ts

```diff
@@ -0,0 +1,17 @@
+import { BlockCascadeRelationOptions, BlockEntity } from './Block.entity';
```

Comment:
Why are you using two entities, `Block.entity & BlockData.entity` instead of a single one that has all the columns?

Reply:
Great question! I was thinking about the buffer as a separate subsystem and haven't really considered a nullable `data` column on the `block` table. Let's consider it, here are my thoughts:

Pros:
- Would save 1 `INSERT` on RollForward, which is great as the projection would be a little bit faster.
- Faster query to select block data for `tip$`.

Cons:
- Slower query to select block data for `tail$`, as you would need a `WHERE` not null. This query is made on initialization and once every 100 (by default) RollForward events.
- If we wanted to support using another implementation of the buffer (not PostgreSQL), but keep the projection sinks, then it'd end up with a trash column that's always null. Not very likely to happen, but possible.

Any more Pros you have in mind @mirceahasegan ?

Reply:
Nope. Thanks for the clarification.

Reply:
Created ADP-2818